### ISSUE TITLE
/run/postgresql and /var/run/postgresql 

### ISSUE THREAD

**Issue by soulne4ny** [docker-library/postgres#242]
[docker-entrypoint.sh#L35](https://github.com/docker-library/postgres/blob/4f3238cf60b514d5c20e86096817718474f53d73/docker-entrypoint.sh#L35) creates `/run/postgresql` while both [Dockerfile-alpine.template#L114](https://github.com/docker-library/postgres/blob/4f3238cf60b514d5c20e86096817718474f53d73/Dockerfile-alpine.template#L114) and [Dockerfile-debian.template#L48](https://github.com/docker-library/postgres/blob/4f3238cf60b514d5c20e86096817718474f53d73/Dockerfile-debian.template#L48) do create `/var/run/postgresql`.

Are both paths used?

**Comment by soulne4ny** [comment]
Are they required at all?

Default postgresql.conf says:
```
/var/lib/postgresql/data # grep pid postgresql.conf
# If external_pid_file is not explicitly set, no extra PID file is written.
#external_pid_file = ''			# write an extra PID file
```

There could be the unix socket .s.PGSQL.5432. But it leaves in `/tmp` with default config:
```
$ docker exec -it 25efac5d8bd0 find / -name .s.\*
/tmp/.s.PGSQL.5432
/tmp/.s.PGSQL.5432.lock
```

**Comment by yosifkit** [comment]
The addition to the Dockerfile was #6 to fix #4, and #205 added to the Dockerfile to make it possible to run with `--tmpfs /run`.  `/var/run` is actually a symlink to `/run`.

I am unable to reproduce not needing it since when I start any `postgres` container it creates the socket file in `/run/postgresql/`.

```console
$ for ver in 9.2 9.3 9.4 9.5 9.6; do docker run -d --name posty "postgres:$ver"; sleep 1; docker exec posty ls -al /run/postgresql/; docker rm -fv posty; done
dad6badf9f4ece26816c22ff6776ee5432d324c716c0ce235481f5e10326d614
total 12
drwxrwsr-x 1 postgres postgres 4096 Dec 20 22:47 .
drwxr-xr-x 1 root     root     4096 Dec 13 23:39 ..
srwxrwxrwx 1 postgres postgres    0 Dec 20 22:47 .s.PGSQL.5432
-rw------- 1 postgres postgres   64 Dec 20 22:47 .s.PGSQL.5432.lock
posty
225eada9dced0b6dd9ae62b4389a1a61a6da080dfad5c30af082f71c98a17d20
total 12
drwxrwsr-x 1 postgres postgres 4096 Dec 20 22:47 .
drwxr-xr-x 1 root     root     4096 Dec 13 23:37 ..
srwxrwxrwx 1 postgres postgres    0 Dec 20 22:47 .s.PGSQL.5432
-rw------- 1 postgres postgres   64 Dec 20 22:47 .s.PGSQL.5432.lock
posty
07068e8170a8852bf2f19e7ab236c6415513af5b12b91765b108e0db8b2b600f
total 12
drwxrwsr-x 1 postgres postgres 4096 Dec 20 22:47 .
drwxr-xr-x 1 root     root     4096 Dec 13 23:35 ..
srwxrwxrwx 1 postgres postgres    0 Dec 20 22:47 .s.PGSQL.5432
-rw------- 1 postgres postgres   64 Dec 20 22:47 .s.PGSQL.5432.lock
posty
1177124e43171733048a7c64aeb9c7614ba0d07bdd009c534fbfc62d10aadc13
total 12
drwxrwsr-x 1 postgres postgres 4096 Dec 20 22:47 .
drwxr-xr-x 1 root     root     4096 Dec 13 23:32 ..
srwxrwxrwx 1 postgres postgres    0 Dec 20 22:47 .s.PGSQL.5432
-rw------- 1 postgres postgres   64 Dec 20 22:47 .s.PGSQL.5432.lock
posty
33703637010e4a7396df4388d85b4b04bf3e66c015e0f7788039fa1abb076138
total 12
drwxrwsr-x 1 postgres postgres 4096 Dec 20 22:47 .
drwxr-xr-x 1 root     root     4096 Dec 13 23:29 ..
srwxrwxrwx 1 postgres postgres    0 Dec 20 22:47 .s.PGSQL.5432
-rw------- 1 postgres postgres   64 Dec 20 22:47 .s.PGSQL.5432.lock
posty
```
